SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosSchedulerClient.java

```diff
@@ -146,7 +147,14 @@ private void connect(URI mesosMasterURI, FrameworkInfo frameworkInfo, Singularit
 
     MesosClientBuilder<Call, Event> clientBuilder = ProtobufMesosClientBuilder.schedulerUsingProtos()
         .mesosUri(mesosMasterURI)
-        .applicationUserAgentEntry(UserAgentEntries.userAgentEntryForMavenArtifact("com.hubspot.singularity", "SingularityService"));
+        .applicationUserAgentEntry(UserAgentEntries.userAgentEntryForMavenArtifact("com.hubspot.singularity", "SingularityService"))
+        .onBackpressureBuffer(
+            scheduler.getEventBufferSize(),
+            () -> {
+              String message = String.format("Overflow of event buffer (%s), singularity could not keep up!", scheduler.getEventBufferSize());
+              scheduler.onUncaughtException(new EventBufferOverflowException(message));
+            },
+            BackpressureOverflow.ON_OVERFLOW_ERROR);
```

Comment:
So just to confirm, if the set of pending events overflows the max buffer size, we get a top-level exception and abort?

Reply:
yes, it'll cause an abort if the buffer overflows. I felt that was more in line with our idea of keep state consistent than dropping things. There is an alternative for using an unbounded buffer, but I figured it'd be better to abort with an exception than to oom

Reply:
Agreed.